Automattic/woocommerce-payments
# Fix using CC token for wrong payment method

Fixes #1635 

#### Changes proposed in this Pull Request


Fix using CC token for wrong payment method


Previously the plugin would default to using the saved CC token during checkout if it existed, regardless of the payment method the customer chose. This PR fixes that issue by only looking for a token that matches the customer's selected payment method. The assumption that if a CC token is available then it should be used was accurate when there was only a single payment method, with the introduction of other payment methods this assumption is no longer accurate. This update now addresses that and will only look for tokens relevant to the payment method the customer chose.

This update also required updating some existing tests to include the `payment_method` key in the request to help with looking up the saved payment method token.





#### Testing instructions


- With a store using Euros and having multiple payment methods enabled (Giropay was the method focused on during testing)
- Perform a checkout using a credit card and save the credit card for future use during checkout
- Begin another checkout, confirm that the saved credit card is selected under the Credit card payment method
- Select Giropay and click Place Order
- Should be able to successfully checkout using Giropay while the saved card was still selected by default under Credit card



**Before**
https://d.pr/i/alps83

**After**
https://d.pr/i/xJ2LtX

-------------------
